src/wp-includes/class-wp-theme-json.php

```diff
@@ -1925,6 +1925,48 @@ public function get_styles_block_nodes() {
 		return static::get_block_nodes( $this->theme_json );
 	}
 
+	/**
+	 * Returns a filtered declarations array if there is a separator block with only a background
+	 * style defined in theme.json by adding a color attribute to reflect the changes in the front.
+	 *
+	 * @since 6.1.1
+	 *
+	 * @param array $declarations List of declarations.
+	 * @return array $declarations List of declarations filtered.
+	 */
+	private static function update_separator_declarations( $declarations ) {
+		$background_matches = array_values(
+			array_filter(
+				$declarations,
+				static function( $declaration ) {
+					return 'background-color' === $declaration['name'];
+				}
+			)
+		);
+
+		if ( isset( $background_matches[0]['value'] ) ) {
+			$border_color_matches = false;
+			$text_color_matches   = false;
+
+			foreach ( $declarations as $declaration ) {
+				if ( 'border-color' === $declaration['name'] ) {
+					$border_color_matches = true;
+				} elseif ( 'color' === $declaration['name'] ) {
+					$text_color_matches = true;
+				}
+			}
+
+			if ( ! $border_color_matches && ! $text_color_matches ) {
+				$declarations[] = array(
+					'name'  => 'color',
+					'value' => $background_matches[0]['value'],
+				);
+			}
+		}
+
+		return $declarations;
```

Comment:
@dream-encode @hellofromtonya This is what I had in mind in regards to my comment above:
```suggestion
		$background_color     = '';
		$border_color_matches = false;
		$text_color_matches   = false;

		foreach ( $declarations as $declaration ) {
			if ( 'background-color' === $declaration['name'] ) {
				$background_color = $declaration['value'];
			} elseif ( 'border-color' === $declaration['name'] ) {
				$border_color_matches = true;
			} elseif ( 'color' === $declaration['name'] ) {
				$text_color_matches = true;
			}

			if ( $background_color && $border_color_matches && $text_color_matches ) {
				break;
			}
		}

		if ( $background_color && ! $border_color_matches && ! $text_color_matches ) {
			$declarations[] = array(
				'name'  => 'color',
				'value' => $background_color,
			);
		}

		return $declarations;
```

Reply:
@felixarntz yeah this is much better than my refactor.  It checks things only when we need to.